lib/block_view.go

```diff
@@ -7166,18 +7504,18 @@ func (bav *UtxoView) _connectAcceptNFTBid(
 	// Sanity check that the royalties are reasonable and won't cause underflow.
 	if txMeta.BidAmountNanos < (creatorRoyaltyNanos + creatorCoinRoyaltyNanos) {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: sum of royalties (%d, %d) is less than bid amount (%d)",
+			"sum of royalties (%d, %d) is less than bid amount (%d)",
 			creatorRoyaltyNanos, creatorCoinRoyaltyNanos, txMeta.BidAmountNanos)
 	}
 
 	bidAmountMinusRoyalties := txMeta.BidAmountNanos - creatorRoyaltyNanos - creatorCoinRoyaltyNanos
-
+	txn, txHash, verifySignatures := txMeta.Txn, txMeta.TxHash, txMeta.VerifySignatures
 	// Connect basic txn to get the total input and the total output without
 	// considering the transaction metadata.
 	totalInput, totalOutput, utxoOpsFromBasicTransfer, err := bav._connectBasicTransfer(
```

Comment:
BUG ALERT: We are not currently checking to see if there was sufficient output to cover a "buy it now" NFT sale.  Right now, one could submit just enough inputs to cover the cost of the txn bytes but not enough inputs to cover the cost of their BuyItNow bid and still have the transaction succeed (except for the fact that the "hardcore balance check" safety net code would probably catch it).  If this is a buy it now NFT, we need to check that totalInput is greater than totalOutput + bid amount.  See the code around "RuleErrorCreatorCoinTxnOutputExceedsInput" for an example.

Reply:
+1. I would put this check in a secon dif statement after the IsAcceptNFT one:
```
if args.TxnType == TxnTypeAcceptNFTBid {
  // As you have it now
} else if args.TxnType == TxnTypeNFTBid {
  // Do the checks for the buy-now NFT
} else {
  // error unexpected txn type
}
```

Reply:
Also don't forget about overflows :)

Reply:
Oh ok I see you added it below. Seems fine, but I would prefer if it was consolidated into the if statement above just to make it so that the checks are in the same place logically.

Reply:
Let's add a test to catch this case also.

Reply:
I'm a little confused about the comment that begins with `I see you added it below`. What is it? the check for overflow or the checks if the Txn is a NFT Bid